### PR TITLE
Fix scrolling in show and tell presentation view

### LINKED ISSUE #224: Disable scrolling in show and tell fullscreen
- Disable the ability to scroll when in fullscreen on show and tell
- Add new click regions at the top/bottom of the screen (where you can just see the next/previous posts), which change post like the existing up/down buttons
- Extra points: When the user tries to scroll up/down, make the click regions and/or the buttons highlighted as a visual prompt

### PR DESCRIPTION
**Pull request by MattIPv4**
Resolves #224

### CODE REVIEW

**Comment by pjeweb** [comment]
Works well, but I do think we would need to make the button more obvious. Either like you suggested when trying to scroll or just make them bigger / more obvious. You can click the next "slide"/post, but they are too small to click, I think.

**Comment by MattIPv4** [comment]
I think I'm going to play around with listening to the wheel event and jumping to the next post if the user scrolls past a threshold -- I see that YT shorts just allow natively scrolling and it is _incredibly_ buggy for me, not an experience I want to replicate.

**Comment by MattIPv4** [comment]
@pjeweb How does this feel to you to use?

**Comment by MattIPv4** [comment]
> but with Chrome and Safari under macOS with a track pad it's scrolling very jerky and does often not scroll to the right end position.

Interesting -- there is no actual scrolling going on here at all, it just calls the scrollIntoView on the element once the scroll event reaches a threshold, so I'm not sure where that jerky motion or misalignment would be coming from

**Comment by pjeweb** [comment]
https://github.com/alveusgg/alveusgg/pull/236

Can you compare how this feels for you? I think the biggest mistake was using snap-mandatory which forces the slide to change even with tiny amounts of wheel scroll distance. This feels decently good in FF and Chrome on Mac. I'm booting windows right now.

**Comment by MattIPv4** [comment]
👀 So in #236, I never get snapped in Firefox, it just scrolls as normal always. No matter how far into a post I scroll, it doesn't snap the alignment ever. (Also, removing the padding/margin has caused the shadows to clip)

**Comment by pjeweb** [comment]
Yeah the shadow clipping should be fixed in none-presentation mode for sure. Was just a quick fix to prevent each slide from having overflow scrolling in the presentation mode.

It does snap for me in FF both on Windows and Mac, but only very very gently if you are within like 100px of the snap point and only if the scroll is decently long and in the right direction. But yeah basically no snapping.

I think that would be the save version so it works everywhere without jenky behavior, but less snappy :)

**Comment by MattIPv4** [comment]
Let me see if I can get snap debouncing working correctly this evening so we get the best of both worlds (free-flowing scroll, and then snapping properly when you stop)

**Comment by pjeweb** [comment]
> Let me see if I can get snap debouncing working correctly this evening so we get the best of both worlds (free-flowing scroll, and then snapping properly when you stop)

Yes please. Let me know if I can help or test anything. It might be worth adding a polyfill for windows/chrome: https://github.com/iamdustan/smoothscroll

**Comment by MattIPv4** [comment]
Does this feel better to you? And did you want me to include the margin/padding removal here, or do you want to do that in a separate PR (ideally that also keeps a fix in place for shadows in the regular view). I started looking at it, but my mind is kinda melting trying to parse how the presentation view actually is laid out

**Comment by pjeweb** [comment]
> Does this feel better to you? 

Hard to tell.

Chrome/Windows is quite bad still because it hard jumps to the snap point (no smoothing). And thats what maya is using. Might be fixed by just throwing in the polyfill.

Otherwise it's not really that much better or worse in Chrome then the live version. But I am biased. Sometimes it's hard to get over the initial scroll threshold and I get snapped back to the old one, sometimes it scrolls to the next/prev with a small scroll distance. Browser scrolling is _hard_. Firefox is behaving well with pretty much all your versions though.

Maybe we need to ask different people for feedback which version they prefer? I really cant tell anymore.

> And did you want me to include the margin/padding removal here, or do you want to do that in a separate PR (ideally that also keeps a fix in place for shadows in the regular view). I started looking at it, but my mind is kinda melting trying to parse how the presentation view actually is laid out

I can tacke that in a separate PR. Yeah its kind of complicated trying to keep the layout. What works for me is just lowering the padding a bit in presentation mode. I cant see the shadow clipping with that:
```typescript
      // Negative margin to allow padding for shadows
      wrapperClassName={`
          flex-1
          ${isPresentationView ? "pb-6 -mb-4" : "pb-16 -mb-12"}
        `}
```

**Comment by MattIPv4** [comment]
> Chrome/Windows is quite bad still because it hard jumps to the snap point (no smoothing). And thats what maya is using. Might be fixed by just throwing in the polyfill.

I tried the polyfill, but that didn't work -- I believe this is re-enabling snapping colliding with the scrollIntoView. Really, I need a way to wait for the scroll to finish, then re-enable snapping. Will look at introducing a dependency to manage that

File: apps/website/src/pages/show-and-tell/index.tsx [hunk 1]
```diff
@@ -122,87 +123,95 @@ const ShowAndTellIndexPage: NextPage<ShowAndTellPageProps> = ({
...
-    start: HTMLElement;
-    distance: number;
+  // Scroll an element into the center of the viewport
+  const scrollTo = useCallback((element: HTMLElement) => {
+    // We use a library to do this, because not every browser supports it
+    // And even in Chrome, smooth breaks for the first/last element of things
+    scrollIntoView(element, {
+      behavior: "smooth",
+      block: "center",
+    });
```
Comment (MattIPv4) [inline apps/website/src/pages/show-and-tell/index.tsx:133]: My only gripe with this now is that this animation-based scrolling has no way to abort if the user starts scrolling again, so you somewhat end up fighting it briefly. I'm not sure that's really that much of a problem though

**Review by pjeweb — APPROVED** [review]

**Pull request merged**